sympy/polys/matrices/lll.py

```diff
@@ -0,0 +1,85 @@
+from math import floor as mfloor
+from typing import Tuple
+
+from sympy.polys.domains import ZZ, QQ
+from sympy.polys.matrices.exceptions import DMRankError, DMShapeError, DMValueError, DMDomainError
+
+logic_error_template = '''FAILURE IN ALGORITHM LOGIC!!!!
+Please file a bug report at https://github.com/sympy/sympy and include this input:
+{}'''
+linear_dependent_error = "input matrix contains linearly dependent rows"
+
+
+def ddm_lll(x, delta=QQ(3, 4)):
+    if QQ(1, 4) >= delta or delta > QQ(1, 1):
+        DMValueError("delta must lie in range (0.25, 1)")
+    if x.shape[0] > x.shape[1]:
+        DMShapeError("input matrix must have shape (m, n) with m <= n")
+    if x.domain != ZZ:
+        DMDomainError("input matrix domain must be ZZ")
```

Comment:
There should be tests that these errors are raised.

I measured coverage with
```
pytest sympy/polys/matrices --cov=sympy.polys.matrices --cov-report=html
```
These are the only lines not covered.

Reply:
I added tests and am raising the exceptions now